src/java.base/share/classes/java/util/ReverseOrderSortedMapView.java

```diff
@@ -0,0 +1,464 @@
+                K key = keyIterator.next();
+                return new ViewEntry<>(map, key, map.get(key));
+            }
+
+            public void remove() {
+                keyIterator.remove();
+            }
+        };
+    }
+
+    static class ViewEntry<K, V> implements Map.Entry<K, V> {
+        final Map<K, V> map;
+        final K key;
+        final V value;
+
+        ViewEntry(Map<K, V> map, K key, V value) {
+            this.map = map;
+            this.key = key;
+            this.value = value;
+        }
+
+        public K getKey()             { return key; }
+        public V getValue()           { return value; }
+        public V setValue(V newValue) { return map.put(key, newValue); }
+
+        public boolean equals(Object o) {
+            return o instanceof Map.Entry<?, ?> e
+                    && Objects.equals(key, e.getKey())
+                    && Objects.equals(value, e.getValue());
+        }
+
+        public int hashCode() {
+            return Objects.hashCode(key) ^ Objects.hashCode(value);
+        }
+
+        public String toString() {
+            return key + "=" + value;
+        }
+    }
+
+    // copied and modified from AbstractMap
+    static <K, V> String toString(Map<K, V> thisMap, Iterator<Entry<K,V>> i) {
+        if (! i.hasNext())
+            return "{}";
+
+        StringBuilder sb = new StringBuilder();
+        sb.append('{');
+        for (;;) {
+            Entry<K,V> e = i.next();
+            K key = e.getKey();
+            V value = e.getValue();
+            sb.append(key   == thisMap ? "(this Map)" : key);
+            sb.append('=');
+            sb.append(value == thisMap ? "(this Map)" : value);
+            if (! i.hasNext())
+                return sb.append('}').toString();
+            sb.append(',').append(' ');
+        }
+    }
+
+    /**
+     * Used for various submap views. We can't use the base SortedMap's subMap,
+     * because of the asymmetry between from-inclusive and to-exclusive.
+     */
+    class Submap extends AbstractMap<K, V> implements SortedMap<K, V> {
+        final K head; // head key, or negative infinity if null
+        final K tail; // tail key, or positive infinity if null
+
+        @SuppressWarnings("unchecked")
+        Submap(K head, K tail) {
+            this.head = head;
+            this.tail = tail;
+        }
+
+        // returns whether e is above the head, inclusive
+        boolean aboveHead(K k) {
+            return head == null || cmp.compare(k, head) >= 0;
+        }
+
+        // returns whether e is below the tail, exclusive
+        boolean belowTail(K k) {
+            return tail == null || cmp.compare(k, tail) < 0;
+        }
+
+        Iterator<Entry<K, V>> entryIterator() {
+            return new Iterator<>() {
+                Entry<K, V> cache = null;
+                K prevKey = null;
+                boolean dead = false;
+                Iterator<Entry<K, V>> it = descendingEntryIterator(base);
```

Comment:
see above about declare it as a local variable, all other fields should be `private`